### PR TITLE
RIA-1154: send decision and reason

### PR DESCRIPTION
**Pull request by t80**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/SendDecisionAndReasonsPdfService.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.service;
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/SendDecisionAndReasonsPdfService.java]: This should probably live in the infrastructure package following our DDD structure.

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/WordDocumentToPdfConverter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import org.apache.commons.io.IOUtils;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.FileType;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.clients.DocmosisDocumentConversionClient;
+
+@Service
+public class WordDocumentToPdfConverter {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/WordDocumentToPdfConverter.java]: Think this is fine, but maybe could be some kind of Converter interface in future.

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -28,6 +30,13 @@ decisionAndReasons.fileExtension: docx
 decisionAndReasons.fileName: "decision-and-reasons-draft"
 decisionAndReasons.templateName: ${IA_DECISION_AND_REASONS_TEMPLATE:TB-IAC-DEC-ENG-00022.docx}
 
+decisionAndReasonsCoverLetter.contentType:  application/pdf
+decisionAndReasonsCoverLetter.fileExtension: PDF
+decisionAndReasonsCoverLetter.fileName: "Decision-and-reasons-Cover-letter"
+decisionAndReasonsCoverLetter.templateName: ${IA_DECISION_AND_REASONS_COVER_LETTER_TEMPLATE:TB-IAC-LET-ENG-00058.docx}
```
Comment (raghera) [inline src/main/resources/application.yaml]: Will have to remember to put this value in AAT and PROD for deployment

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/WordDocumentToPdfConverter.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    public WordDocumentToPdfConverter(DocmosisDocumentConversionClient docmosisDocumentConversionClient) {
+        this.docmosisDocumentConversionClient = docmosisDocumentConversionClient;
+    }
+
+    public File convertResourceToPdf(Resource resource) {
+
+        File tempPdfFile;
+
+        try {
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/WordDocumentToPdfConverter.java]: You can use try-with-resource block

**Pull request merged**